crates/matrix-sdk-ui/src/room_list_service/room_list.rs

```diff
@@ -142,8 +144,41 @@ impl RoomList {
         let stream = stream! {
             loop {
                 let filter_fn = filter_fn_cell.take().await;
-                let (values, stream) = list
-                    .room_list_stream()
+                let (raw_values, mut raw_stream) = list.room_list_stream();
+                let mut raw_current_values = raw_values.clone();
+                let client = client.clone();
+                let raw_stream_with_recv = stream! {
+                    loop {
+                        let mut roominfo_update_recv = client.roominfo_update_recv();
+                        select! {
+                            v = raw_stream.next() => {
+                                if let Some(v) = v {
+                                    for change in &v {
+                                        change.clone().apply(&mut raw_current_values);
+                                    }
+                                    yield v;
+                                } else {
+                                    break;
+                                }
+                            }
+                            room_id = roominfo_update_recv.recv() => {
+                                if let Ok(room_id) = room_id {
```

Comment:
style: I think you can rewrite it like that, to spare one level of indent:

```suggestion
                            Ok(room_id) = roominfo_update_recv.recv() => {
```